code/backend/pages/01_Ingest_Data.py

```diff
@@ -75,21 +77,32 @@ def remote_convert_files_and_add_embeddings(process_all=False):
 
 
 def add_urls():
-    params = {}
-    if env_helper.FUNCTION_KEY is not None:
-        params["code"] = env_helper.FUNCTION_KEY
-        params["clientId"] = "clientKey"
     urls = st.session_state["urls"].split("\n")
-    for url in urls:
-        body = {"url": url}
-        backend_url = urllib.parse.urljoin(
-            env_helper.BACKEND_URL, "/api/AddURLEmbeddings"
-        )
-        r = requests.post(url=backend_url, params=params, json=body)
-        if not r.ok:
-            raise ValueError(f"Error {r.status_code}: {r.text}")
-        else:
-            st.success(f"Embeddings added successfully for {url}")
+    if env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
+        try:
+            for url in urls:
+                response = requests.get(url)
+                parsed_data = BeautifulSoup(response.content, "html.parser")
+                with io.BytesIO(parsed_data.get_text().encode("utf-8")) as stream:
+                    upload_file(stream, url)
```

Comment:
Have you checked the the URL is stored correctly? I know there's some post processing that happens for it.

(though just noticed the metadata you're adding so probably all good)

Reply:
Yes verified, it is uploading the url data correctly. In case of push, it does all the chunking, processing once url is added. In case of pull, we only need it to get indexed. Let me know if I am missing anything